docs/tutorials/go.md

```diff
@@ -150,10 +152,34 @@ func (KVStoreApplication) ApplySnapshotChunk(abcitypes.RequestApplySnapshotChunk
 }
 ```
 
-Now I will go through each method explaining when it's called and adding
+Now, we will go through each method and explain when it is executed while adding
 required business logic.
 
-### 1.3.1 CheckTx
+### 1.3.1 Key-value store setup
+
+For the underlying key-value store we'll use the latest version of [badger](https://github.com/dgraph-io/badger), which is an embeddable, persistent and fast key-value (KV) database.
+
+```sh
+$ go get github.com/dgraph-io/badger/v3
+go: added github.com/dgraph-io/badger/v3 v3.2103.2
+```
+
+```go
+import "github.com/dgraph-io/badger/v3"
+
+type KVStoreApplication struct {
+ db           *badger.DB
+ currentBatch *badger.Txn
+}
+
+func NewKVStoreApplication(db *badger.DB) *KVStoreApplication {
+ return &KVStoreApplication{
+  db: db,
+ }
+}
+```
+
+### 1.3.2 CheckTx
```

Comment:
Would be nice to mention the notion of `priority` and `sender` here if possible

Reply:
Hi @cmwaters, I am still in the process of exploring the Tendermint ecosystem so I am not sure if I can document that properly. I was only aiming to update the Go tutorials so others can also benefit from what I learned so far.

Still, if you require it, I would try to explore about this topic and add it to the PR.

Reply:
That's okay - we can just file an issue and follow up at a later point

Reply:
Got it, thanks @cmwaters!